screens/LoginScreen.tsx

```diff
@@ -39,7 +40,20 @@ export default function LoginScreen({ navigation }: ScreenWithNavigation) {
     if (response?.type === "success") {
       const { id_token } = response.params;
       const credential = GoogleAuthProvider.credential(id_token);
-      signInWithCredential(authContext.auth, credential);
+      signInWithCredential(authContext.auth, credential)
+        .then(async ({ user }) => {
+          return getUser(user);
+        })
+        .then(
+          ({ username }) => {
+            authContext.setUserName(username);
+          },
+          (e) => {
+            if (e instanceof BadStatusError && e.code == 404) {
+              navigation.navigate("Register", { fromGoogleFlow: true });
+            }
```

Comment:
Yeah, I don't know how best to write this code... But basically, here we have succesfulyl authed from google but haven't yet set up our account on the backend (because a 404 when we try to get a userdetails from the firebase id), so we have to set up a username and create a user entry in the backend

Reply:
I see, maybe when someone goes through the google flow we just user their email as the username? I feel like thats how I've seen it done in other apps